Source/Core/Common/BitFieldView.h

```diff
@@ -0,0 +1,788 @@
+{
+  return ConstBitFieldFixedView<field_t, start, width, host_t>(host);
+}
+template <typename field_t, typename host_t>
+constexpr BitFieldView<field_t, host_t>  //
+MakeBitFieldView(const std::size_t start, const std::size_t width, host_t& host)
+{
+  return BitFieldView<field_t, host_t>(start, width, host);
+}
+template <typename field_t, typename host_t>
+constexpr ConstBitFieldView<field_t, host_t>  //
+MakeConstBitFieldView(const std::size_t start, const std::size_t width, const host_t& host)
+{
+  return ConstBitFieldView<field_t, host_t>(start, width, host);
+}
+template <typename field_t, std::size_t start, std::size_t width, std::size_t length,
+          typename host_t>
+constexpr BitFieldFixedViewArray<field_t, start, width, length, host_t>  //
+MakeBitFieldFixedViewArray(host_t& host)
+{
+  return BitFieldFixedViewArray<field_t, start, width, length, host_t>(host);
+}
+template <typename field_t, std::size_t start, std::size_t width, std::size_t length,
+          typename host_t>
+constexpr ConstBitFieldFixedViewArray<field_t, start, width, length, host_t>  //
+MakeConstBitFieldFixedViewArray(const host_t& host)
+{
+  return ConstBitFieldFixedViewArray<field_t, start, width, length, host_t>(host);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename field_t_, std::size_t start_, std::size_t width_, typename host_t_>
+class BitFieldFixedView
+{
+public:
+  using host_t = host_t_;
+  using field_t = field_t_;
+  constexpr static std::size_t start = start_;
+  constexpr static std::size_t width = width_;
+
+private:
+  host_t& host;
+
+public:
+  BitFieldFixedView() = delete;
+  constexpr BitFieldFixedView(host_t& host_) : host(host_){};
+
+  constexpr field_t Get() const { return GetBitFieldFixed<field_t, start, width>(host); }
+  constexpr void Set(const field_t val) { SetBitFieldFixed<field_t, start, width>(host, val); }
+
+  constexpr BitFieldFixedView& operator=(const BitFieldFixedView& rhs)
+  {
+    return operator=(rhs.Get());
+  }
+  constexpr BitFieldFixedView& operator=(const field_t rhs)
+  {
+    Set(rhs);
+    return *this;
+  }
+  constexpr BitFieldFixedView& operator+=(const field_t rhs) { return operator=(Get() + rhs); }
+  constexpr BitFieldFixedView& operator-=(const field_t rhs) { return operator=(Get() - rhs); }
+  constexpr BitFieldFixedView& operator*=(const field_t rhs) { return operator=(Get() * rhs); }
+  constexpr BitFieldFixedView& operator/=(const field_t rhs) { return operator=(Get() / rhs); }
+  constexpr BitFieldFixedView& operator|=(const field_t rhs) { return operator=(Get() | rhs); }
+  constexpr BitFieldFixedView& operator&=(const field_t rhs) { return operator=(Get() & rhs); }
+  constexpr BitFieldFixedView& operator^=(const field_t rhs) { return operator=(Get() ^ rhs); }
+
+  constexpr operator field_t() const { return Get(); }
+  constexpr bool operator!() const { return !static_cast<bool>(Get()); }
```

Comment:
Perhaps also consider a `constexpr explicit operator bool() const` you feel this may be helpful.

Reply:
I know [why you suggest that](https://en.cppreference.com/w/cpp/language/operators#Boolean_negation_operator), but adding an explicit operator bool() causes a duplicate definition with BitFieldViews of a bool field_t.